bindings/python/iota_sdk/types/unlock.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2023 IOTA Stiftung
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+from dataclasses import dataclass
+from enum import IntEnum
+from typing import Optional
+from iota_sdk.types.signature import Ed25519Signature
+
+
+class UnlockType(IntEnum):
+    Signature = 0
+    Reference = 1
+    Alias = 2
+    Nft = 3
+
+
+@dataclass
+class Unlock:
+    """Unlock type.
+    """
+    type: int
+    signature: Optional[Ed25519Signature] = None
+    reference: Optional[int] = None
+
+    def as_dict(self):
+        config = {k: v for k, v in self.__dict__.items() if v is not None}
+
+        if 'signature' in config:
+            config['signature'] = config['signature'].__dict__
+
+        return config
+
+
+@dataclass
+class SignatureUnlock:
+    """An unlock holding a signature unlocking one or more inputs.
+    """
+    signature: Ed25519Signature
```

Comment:
Should technically be a `Signature`, maybe we don't have that yet. But if `Ed25519Signature` serialises it's type it's fine.

Reply:
Will do later?

Reply:
Added it to the issue